django_ai_assistant/langchain/chat_message_histories.py

```diff
@@ -73,13 +77,17 @@ def add_messages(self, messages: Sequence[BaseMessage]) -> None:
             messages: A list of BaseMessage objects to store.
         """
         with transaction.atomic():
+            existing_messages = self.get_messages()
+
+            messages_to_create = [m for m in messages if m not in existing_messages]
```

Comment:
how is the `not in` making the equality for messages? do you know?

Reply:
Good point, it worked but I didn't check why it was working, perhaps we should force comparing by id to make this safer and more explicit?

Reply:
The reason why this works is mainly because the built in `add_messages` function that is called before this method is called does the heavy lifting of returning just the messages that are new. So the code here is actually wrong, I'll return it to the previous state so there's no risk of someone thinking this is doing more than it actually is.

Reply:
I think so, but does the messages Langchain passes as params contain any ids?
Also, we can optimize this a bit by querying with ids.

Reply:
The langchain ones receive a uuid id that is added in the built in `add_messages` function that is called before calling this one and they are update here to the db ids a few lines bellow this.

Reply:
I've made the change to compare ids and optimized the query to only return the ids